src/v/cluster/archival/tests/ntp_archiver_reupload_test.cc

```diff
@@ -414,7 +414,7 @@ FIXTURE_TEST(test_upload_compacted_segments_concat, reupload_fixture) {
     BOOST_REQUIRE_EQUAL(replaced[1].base_offset, model::offset{1000});
 
     verify_concat_segment_request(
```

Comment:
Why we don't have an assert on `manifest.size() == 1`?

Reply:
replaced this with the size check